### DISCUSSION TITLE
Ability to output the detected text

### DISCUSSION THREAD

**Discussion by cohsieh23** [microsoft/presidio#609] [Requested features]
It would be nice to have the ability to get the text that is being detected.
Currently, I have to export the results to a dictionary, then use the start/end indices to pull out the text myself.

**Comment by navalev** [comment]
Hi @cohsieh23, thanks for the feedback. I assume you mean add the detected text to the analyzer response, we will consider this in our rodemap and this could be a great addition by the cummunity.

**Reply by Onenguyen** [reply]
Hi! I'm just getting started and my use case requires me to extract the detected text. Is this going to be integrated soon?

**Reply by omri374** [reply]
Hi @cohsieh23 and @Onenguyen, please see the samplein this notebook:  https://github.com/microsoft/presidio/blob/main/docs/samples/python/getting_entity_values.ipynb

Any feedback is greatly appreciated.

**Reply by Onenguyen** [reply]
This looks great! I'm pretty new to this process, but is this available now or will it be in upcoming releases?

**Reply by omri374** [reply]
The sample is available now. It essentially shows how to use existing parts in Presidio to achieve what you need.

**Reply by Onenguyen** [reply]
@omri374 This is working great! I'm stuck on trying to figure out how to include the entity so I have more context to what is being detected.

`[(cleantext[res.start:res.end], res.start, res.end) for res in results]`

```
('CHEE LO VANG   ', 126, 141)
```
I'd like

`[('PERSON'. 'CHEE LO VANG   ', 126, 141)]`

**Reply by omri374** [reply]
This should work:
`[(res.entity_type, cleantext[res.start:res.end], res.start, res.end) for res in results]`

**Comment by josiahrohrer** [comment]
I have a similar, but slightly different problem and stumbled across this page. I defined a custom anonymization rule that counts up the entities, so first person gets mapped to <PERSON_1> second person to <PERSON_2> and i wish to also have a store which word got mapped to which entity so something like {<PERSON_1>: "Max Mueller", <PERSON_2>: "Karl"}. Does anybody know how i can achieve this. The problem is in the result from the analyzer I dont know to what it gets anonymized and in the result of the anonymizer I cannot seem to access the original text that was anonymized.
Thanks a lot for your help. 
Fyi the result from the anonymizer and the ones from the analyzer have a different ordering

**Reply by omri374** [reply]
@josiahrohrer the sample in #1284 has such mapping stored. It's still being reviewed but should be merged in the next few days.
